pkg/virt-handler/isolation/isolation.go

```diff
@@ -354,37 +354,35 @@ func (r *realIsolationResult) IsBlockDevice(path string) (bool, error) {
 // entry in /proc/mountinfo of the isolation result of the given process.
 func (r *realIsolationResult) ParentMountInfoFor(mountInfo *MountInfo) (parentMountInfo *MountInfo, err error) {
 	if err = forEachRecord(r.mountInfo(), func(record []string) bool {
-		if record[2] == mountInfo.DeviceContainingFile {
-			parentMountInfo = &MountInfo{
-				DeviceContainingFile: record[2],
-				Root:                 record[3],
-				MountPoint:           record[4],
-			}
+		if record[2] != mountInfo.DeviceContainingFilesystem ||
+			!strings.HasPrefix(mountInfo.PathOnFilesystem, record[3]) ||
+			(parentMountInfo != nil && len(record[3]) <= len(parentMountInfo.PathOnFilesystem)) {
```

Comment:
I guess `parentMountInfo != nil` will always evaluate to `false` here since it is assigned bellow before breaking the iteration.

Reply:
What do you mean? If the device didn't match or the path on the filesystem on the device doesn't match, it will stay `nil` until both the device and path have been found.  
From that point on, if we find other entries that match the device an path on the filesystem we should only replace `parentMountInfo` if we found a longer match.

You are however correct in that it breaks the iteration after the first find/match. I thought the function was akin to a filter predicate but it aborts parsing the rest of the entries/records when `true` is returned. I should drop that `return true` otherwise if there is a longer match available it would never be found.

Reply:
Updated to always return false since I want to match/check all records, if statement is the same as before but inverted to not duplicate the return statement.